src/sentry/tasks/groupowner.py

```diff
@@ -2,100 +2,89 @@
+                    project, group_id, event_frames, event_platform
+                )
+            owner_scores = {}
+            for committer in committers:
+                if "id" in committer["author"]:
+                    author_id = committer["author"]["id"]
+                    for commit, score in committer["commits"]:
+                        if score >= MIN_COMMIT_SCORE:
+                            owner_scores[author_id] = max(score, owner_scores.get(author_id, 0))
 
-                    if owner_scores:
-                        for owner_id in sorted(owner_scores, reverse=True, key=owner_scores.get)[
-                            :PREFERRED_GROUP_OWNERS
-                        ]:
-                            go, created = GroupOwner.objects.update_or_create(
-                                group_id=event.group_id,
-                                type=GroupOwnerType.SUSPECT_COMMIT.value,
-                                user_id=owner_id,
-                                project=project,
-                                organization_id=project.organization_id,
-                                defaults={
-                                    "date_added": timezone.now()
-                                },  # Updates date of an existing owner, since we just matched them with this new event
-                            )
-                            if created:
-                                owner_count += 1
-                                if owner_count > PREFERRED_GROUP_OWNERS:
-                                    owners.first().delete()
-                except Commit.DoesNotExist:
-                    logger.info(
-                        "process_suspect_commits.skipped",
-                        extra={"event": event.event_id, "reason": "no_commit"},
-                    )
-                except Release.DoesNotExist:
-                    logger.info(
-                        "process_suspect_commits.skipped",
-                        extra={"event": event.event_id, "reason": "no_release"},
+            if owner_scores:
+                for owner_id in sorted(owner_scores, reverse=True, key=owner_scores.get)[
+                    :PREFERRED_GROUP_OWNERS
+                ]:
+                    go, created = GroupOwner.objects.update_or_create(
+                        group_id=group_id,
+                        type=GroupOwnerType.SUSPECT_COMMIT.value,
+                        user_id=owner_id,
+                        project=project,
+                        organization_id=project.organization_id,
+                        defaults={
+                            "date_added": timezone.now()
+                        },  # Updates date of an existing owner, since we just matched them with this new event
                     )
+                    if created:
+                        owner_count += 1
+                        if owner_count > PREFERRED_GROUP_OWNERS:
+                            owners.first().delete()
```

Comment:
Do you intentionally want .first()? Or are you trying to use a shortcut for "one row"?

Reply:
I think both.

It is sorted by date so I'd want the first row from that queryset, which would be the "oldest". Do you think there is a better way to do this?

Reply:
Nope, if there's already a sort, that's fine. Just keep in mind, this may return None and error because of that. I know you're guarding this around a .exists() but in theory is still possible. Up to you if you wanna handle this case.

Reply:
I made a note to follow up with this in another PR :)